### DISCUSSION TITLE
Generate a HSS dat file to use with DirectC

### DISCUSSION THREAD

**Discussion by rary21** [polarfire-soc/polarfire-soc-documentation#479] [HSS and U-Boot] [answered]
Can I generate a HSS dat file to use with DirectC?
In build directory, I can find several binaries that can be set as ENVM data on Libero SoC.
However, I want to flash HSS only via SPI-DirectC.

**Comment by hughbreslin** [comment, marked as answer]
Hey @rary21 you could probably set up a custom version of the boot mode programmer to export the dat file for you but the easiest route might be just using Libero SoC directly - you can add the HSS as a design client for the eNVM and export a DAT file containing only the eNVM content. For example:

Adding the client can be done here:
![image](https://github.com/user-attachments/assets/84f53caa-b50b-4a12-b1a9-919402dd11d4)

and the dat file can be generated here:
![image](https://github.com/user-attachments/assets/7bd2685e-3826-4214-9b6f-fab3f879bc81)

You can specify that only the eNVM will be programmed by deselecting the un-needed options

**Reply by rary21** [reply]
Thank you !
